Source/Core/Core/AchievementManager.cpp

```diff
@@ -457,6 +511,7 @@ void AchievementManager::Shutdown()
   if (m_client)
   {
     CloseGame();
+    std::lock_guard lg{m_lock};
```

Comment:
Locking here can lead to a deadlock. You can't know how many events the queue will still process after you lock this but before it processes the Shutdown request, which can lead to the situation where the queue waits for the lock to become free and this thread waits for the queue to shut down. Move the lock after the Shutdown().